component/src/main/java/org/wso2/extension/siddhi/execution/geo/GetCityResolverFunction.java

```diff
@@ -0,0 +1,216 @@
+     *             runtime data value of the Function parameter
+     * @return the Function result
+     */
+    @Override
+    protected Object execute(Object data) {
+        String city = null;
+        if (isCacheEnabled) {
+            city = cityLRUCache.get(data.toString());
+        }
+        if (city == null) {
+            city = geoLocationResolverImpl.getCity(data.toString());
+            if (city != null) {
+                cityLRUCache.put(data.toString(), city);
+            } else {
+                cityLRUCache.put(data.toString(), "");
+            }
+        }
+        return city != null ? city : "";
+    }
+
+    /**
+     * return a Class object that represents the formal return type of the method represented by this Method object.
+     *
+     * @return the return type for the method this object represents
+     */
+    @Override
+    public Attribute.Type getReturnType() {
+        return Attribute.Type.STRING;
+    }
+
+    /**
+     * Used to collect the serializable state of the processing element, that need to be
+     * persisted for reconstructing the element to the same state on a different point of time
+     *
+     * @return stateful objects of the processing element as an map
+     */
+    @Override
+    public Map<String, Object> currentState() {
+        return null;
+    }
+
+    /**
+     * Used to restore serialized state of the processing element, for reconstructing
+     * the element to the same state as if was on a previous point of time.
+     *
+     * @param state the stateful objects of the processing element as a map.
+     *              This is the same map that is created upon calling currentState() method.
+     */
+    @Override
+    public void restoreState(Map<String, Object> state) {
+
+    }
+
+    private void initializeExtensionConfigs(ConfigReader configReader) throws SiddhiAppValidationException {
+        String geoResolverImplClassName = configReader.readConfig("geoLocationResolverClass",
+                DEFAULT_GEOLOCATION_RESOLVER_CLASSNAME);
+        isCacheEnabled = Boolean.parseBoolean(configReader.readConfig(IS_CACHE_ENABLED, "true"));
+        String configCacheValue = configReader.readConfig(CACHE_SIZE_KEY, String.valueOf(cacheSize));
+        if (isCacheEnabled) {
+            try {
+                cacheSize = Integer.parseInt(configCacheValue);
+            } catch (NumberFormatException e) {
+                cacheSize = DEFAULT_CACHE_SIZE;
+                log.warn("The config '" + configCacheValue + "' provided in 'cacheSize' is not a valid integer. " +
+                        "Hence using the default cache size '" + DEFAULT_CACHE_SIZE + "'");
+            } finally {
+                cityLRUCache = new LRUCache<>(cacheSize);
+            }
+        }
+
+        try {
+            geoLocationResolverImpl = (GeoLocationResolver) Class.forName(geoResolverImplClassName).newInstance();
+            geoLocationResolverImpl.init(configReader);
```

Comment:
also, need to have shutdown method in the interface to handle cleaning tasks if any